tests/Unit/ApparentHorizons/Test_Strahlkorper.cpp

```diff
@@ -0,0 +1,457 @@
+    const double r = 3.0;
+    Strahlkorper<Frame::Inertial> s(4, 4, r, center);
+    CHECK(s.average_radius() == approx(r));
+  }
+
+  // Create a strahlkorper with a Im(Y11) dependence.
+  const size_t l_max = 4, m_max = 4;
+  const double radius = 2.0;
+  const double y11_amplitude = 1.0;
+  const std::array<double, 3> center = {{0.1, 0.2, 0.3}};
+
+  Strahlkorper<Frame::Inertial> strahlkorper_sphere(l_max, m_max, radius,
+                                                    center);
+  auto coefs = strahlkorper_sphere.coefficients();
+  SpherepackIterator it(l_max, m_max);
+  // Conversion between SPHEREPACK b_lm and real valued harmonic coefficients:
+  // b_lm = (-1)^{m+1} sqrt(1/2pi) d_lm
+  coefs[it.set(1, -1)()] = y11_amplitude * sqrt(0.5 / M_PI);
+  Strahlkorper<Frame::Inertial> strahlkorper(strahlkorper_sphere, coefs);
+
+  // Now construct a Y00 + Im(Y11) surface by hand.
+  const auto& theta_phi = strahlkorper.ylm_spherepack().theta_phi_points();
+  const auto& theta_points = strahlkorper.ylm_spherepack().theta_points();
+  const auto& phi_points = strahlkorper.ylm_spherepack().phi_points();
+  const auto n_pts = theta_phi[0].size();
+  YlmTestFunctions::Y11 func;
+  DataVector test_radius(n_pts);
+  func.func(&test_radius, 1, 0, theta_points, phi_points);
+  for (size_t s = 0; s < n_pts; ++s) {
+    test_radius[s] *= y11_amplitude;
+    test_radius[s] += radius;
+  }
+
+  // Create DataBox
+  auto box =
+      db::create<db::AddTags<StrahlkorperDB::TagList<Frame::Inertial>::Tags>,
+                 db::AddComputeItemsTags<StrahlkorperDB::TagList<
+                     Frame::Inertial>::ComputeItemsTags>>(strahlkorper);
+
+  // Test radius
+  const auto& strahlkorper_radius =
+      db::get<StrahlkorperDB::Radius<Frame::Inertial>>(box);
+  for (size_t s = 0; s < n_pts; ++s) {
+    CHECK(strahlkorper_radius[s] == approx(test_radius[s]));
+  }
+
+  // Test derivative of radius
+  StrahlkorperDB::types<Frame::Inertial>::OneForm test_dx_radius(n_pts);
+  for (size_t s = 0; s < n_pts; ++s) {
+    // Analytic solution I computed in Mathematica
+    const double theta = theta_phi[0][s];
+    const double phi = theta_phi[1][s];
+    const double r = test_radius[s];
+    const double sin_phi = sin(phi);
+    const double cos_phi = cos(phi);
+    const double sin_theta = sin(theta);
+    const double cos_theta = cos(theta);
+    const double amp = -sqrt(3.0 / 8.0 / M_PI) * y11_amplitude;
+
+    test_dx_radius.get(0)[s] = -((cos_phi * sin_phi) / r) +
+                               (cos_phi * square(cos_theta) * sin_phi) / r;
+    test_dx_radius.get(1)[s] =
+        square(cos_phi) / r + (square(cos_theta) * square(sin_phi)) / r;
+    test_dx_radius.get(2)[s] = -((cos_theta * sin_phi * sin_theta) / r);
+    for (auto& a : test_dx_radius) {
+      a[s] *= amp;
+    }
+  }
+  const auto& strahlkorper_dx_radius =
+      db::get<StrahlkorperDB::DxRadius<Frame::Inertial>>(box);
+  for (size_t i = 0; i < 3; ++i) {
```

Comment:
try CHECK_ITERABLE_APPROX

Reply:
The macro can handle nesting, so as long as `decltype(strahlkorper_dx_radius)` is iterable, you shouldn't need the loop, and since it looks like it's a tensor it should be iterable.